src/components/StylesFromLiteralsService.js

```diff
@@ -59,32 +61,65 @@ goog.provide('ga_stylesfromliterals_service');
         var olStyles = {};
         var style = value.vectorOptions;
         var geomType = value.geomType;
+
         if (geomType === 'point') {
-          style = {
-            image: style
-          };
-        }
-        angular.forEach(style, function(value, key) {
-          var olStyle = {};
-          if (key === 'image') {
-            var styleP = style[key];
-            var options = getOlBasicStyles(styleP);
-            // {} to preserve the original object
-            options = angular.extend({}, styleP, options);
-            olStyle = getOlStyleForPoint(options, value.type);
-            olStyles[key] = olStyle;
-          } else {
-            olStyles = angular.extend({}, olStyle, getOlBasicStyles(style));
+          var olText;
+          if (style.label) {
+            olText = getOlBasicStyles(style.label).text;
           }
-        });
+          var basicStyles = getOlBasicStyles(style);
+          var olImage = angular.extend({}, style, basicStyles);
+          delete olImage.label;
+          olImage = getOlStyleForPoint(olImage, style.type);
+          olStyles.image = olImage;
+          olStyles.text = olText;
+        } else {
+          angular.forEach(style, function(value, key) {
+            if (key === 'label') {
+              olStyles['text'] = getOlBasicStyles(style[key])['text'];
+            } else if (['stroke', 'fill', 'image'].indexOf(key) !== -1) {
+              olStyles[key] = getOlBasicStyles(style)[key];
+            }
+          });
+        }
         return new ol.style.Style(olStyles);
       }
 
-      function olStyleForPropertyValue(properties) {
-        var olStyle;
+      function getGeomTypeFromGeometry(olGeometry) {
+        if (olGeometry instanceof ol.geom.Point ||
+            olGeometry instanceof ol.geom.MultiPoint) {
+          return 'point';
+        } else if (olGeometry instanceof ol.geom.LineString ||
+            olGeometry instanceof ol.geom.MultiLineString) {
+          return 'line';
+        } else if (olGeometry instanceof ol.geom.Polygon ||
+            olGeometry instanceof ol.geom.MultiPolygon) {
+          return 'polygon';
+        }
+      }
 
+      function getLabelProperty(value) {
+        if (value) {
+          return value.property !== undefined ? value.property : null;
+        }
+        return null;
```

Comment:
Any reason to return null value ? 

If undefined ois also correct you can simplify this line to : 
```
if (value) {
  return value.property;
}
```